### PR TITLE
support `EXIF_TAG_USER_COMMENT`

### PR DESCRIPTION
**Pull request by t-bltg**
Tests added, adding `EXIF_TAG_MAKER_NOTE` is left for another PR.

I also took the liberty of updating some docstrings, and the README.md.

Fix https://github.com/JuliaImages/ExifViewer.jl/issues/24.

### CODE REVIEW

File: src/write.jl [hunk 1]
```diff
@@ -39,33 +49,29 @@ function set_value(ptrentry, tagv)
...
-        unsafe_copyto!(buf, pointer(Vector{UInt8}(tagv * "\0")), len)
-        ptrentry.data = buf
+        ptrentry.data = get_ascii_buffer(ptrentry, tagv)
     elseif entry.format == LibExif.EXIF_FORMAT_SRATIONAL
         p = Rational(parse(Float32, tagv))
         LibExif.exif_set_srational(entry.data, FILE_BYTE_ORDER, LibExif.ExifSRational(p.num, p.den))
     elseif entry.format == LibExif.EXIF_FORMAT_UNDEFINED
         if entry.tag == LibExif.EXIF_TAG_FLASH_PIX_VERSION
             data = Dict{String,String}(
-            "FlashPix Version 1.0" => "0100\0",
```
Comment (t-bltg) [inline src/write.jl:62]: Some indentation consistency.

File: src/write.jl [hunk 2]
```diff
@@ -39,33 +49,29 @@ function set_value(ptrentry, tagv)
     elseif entry.format == LibExif.EXIF_FORMAT_LONG
         LibExif.exif_set_long(entry.data, FILE_BYTE_ORDER, parse(Cuint, tagv))
     elseif entry.format == LibExif.EXIF_FORMAT_RATIONAL
-        if entry.tag in [LibExif.EXIF_TAG_FNUMBER, LibExif.EXIF_TAG_APERTURE_VALUE, LibExif.EXIF_TAG_MAX_APERTURE_VALUE]    
+        if entry.tag in (LibExif.EXIF_TAG_FNUMBER, LibExif.EXIF_TAG_APERTURE_VALUE, LibExif.EXIF_TAG_MAX_APERTURE_VALUE)
```
Comment (t-bltg) [inline src/write.jl:52]: It's better to use `Tuple`s instead of `Vector`s here (size inferred at compile time).

File: README.md [hunk 3]
```diff
@@ -50,18 +50,18 @@ julia> using TestImages, ExifViewer
 julia> filepath = testimage("earth_apollo17.jpg", download_only=true)
 julia> io = open(filepath, "r")
 julia> read_tags(io; read_all=false, tags=["EXIF_TAG_FLASH_PIX_VERSION", "EXIF_TAG_ORIENTATION"])
-Dict{Any, Any} with 2 entries:
+Dict{String, String} with 2 entries:
```
Comment (ashwani-rathee) [inline README.md:53]: I should have fixed these in previous PRs where I switched from `Dict{Any, Any}` to `Dict{String, String}` for performance. Thanks!

File: src/write.jl [hunk 4]
```diff
@@ -39,33 +49,29 @@ function set_value(ptrentry, tagv)
...
+                "FlashPix Version 1.0" => "0100\0",
+                "FlashPix Version 1.01" => "0101\0",
+                "Unknown FlashPix Version" => "0000\0",
             )
             unsafe_copyto!(entry.data, pointer(Vector{UInt8}(data[tagv])), 5)
+        elseif entry.tag == LibExif.EXIF_TAG_USER_COMMENT
+            ptrentry.data = get_ascii_buffer(ptrentry, tagv)
         else
-            @debug "Tag unsupported" entry.tag
+            @debug "Tag unsupported (EXIF_FORMAT_UNDEFINED)" entry.tag
```
Comment (ashwani-rathee) [inline src/write.jl:74]: Better debug messages! albeit I wonder should I make these `@info` to inform of unsupported tags. This caused an issue for you I think where you couldn't identify where that comment tag was disappearing.

File: test/write.jl [hunk 5]
```diff
@@ -34,10 +34,12 @@
...
         "EXIF_TAG_CONTRAST"=>"Normal",
         "EXIF_TAG_SHARPNESS"=>"Normal",
-        "EXIF_TAG_COMPONENTS_CONFIGURATION"=> "Y Cb Cr -"
+        "EXIF_TAG_COMPONENTS_CONFIGURATION"=> "Y Cb Cr -",
+        "EXIF_TAG_USER_COMMENT"=>"Dummy comment",
+        # "EXIF_TAG_MAKER_NOTE"=>"Maker Note",  # TODO: support `EXIF_TAG_MAKER_NOTE`
     )
     path = joinpath(tempdir(), "tmp.jpg")
-    write_tags(path; img, tags=tags)
+    write_tags(path; img, tags)
```
Comment (ashwani-rathee) [inline test/write.jl:42]: Any opinion on tests?  how could we make them more robust?

**Review by ashwani-rathee — APPROVED** [review]
Looks good to me!

**Comment by ashwani-rathee** [comment]
BTW is writing `EXIF_TAG_MAKER_NOTE` important for you? We can read them right now just fine but writing them is still needed to be done.

**Comment by t-bltg** [comment]
> BTW is writing EXIF_TAG_MAKER_NOTE important for you? We can read them right now just fine but writing them is still needed to be done.

Not at the moment, but I though that integrating more tags was always better. This is left fo another PR :)

File: test/write.jl (hunk 5 above)
Comment (t-bltg) [reply test/write.jl:42]: I think there currently are ok, tests are always incremental :)

File: src/write.jl (hunk 4 above)
Comment (t-bltg) [reply src/write.jl:74]: I should have read the source code first and set `JULIA_DEBUG=ExifViewer` when hitting  https://github.com/JuliaImages/ExifViewer.jl/issues/24 ...
It's always a tradeoff for end users not wanting spurious warnings.
But silently ignoring is confusing too, so I don't have a strong opinion on that one (appart from disambiguating the `@debug "Tag unsupported"` message from the one a few lines below).

**Pull request merged**